hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -54,6 +58,37 @@
 @InterfaceAudience.Private
 public class MetricsConnection implements StatisticTrackable {
 
+  static final Map<String, MetricsConnection> METRICS_INSTANCES =
+    new HashMap<String, MetricsConnection>();
+
+  static MetricsConnection getMetricsConnection(final String scope,
```

Comment:
The comment on line 55-56 about "be sure to call shutdown()" could be removed, and replaced by "Calling `getMetricsConnection` implicity creates and "starts" instances of these classes; be sure to call `deleteMetricsConnection` to terminate the thread pools they allocate. You could also add a comment about the ref counting model now being used for these objects.

Reply:
Great catch, updated.